e2e/nomostest/client.go

```diff
@@ -110,16 +112,60 @@ type Cluster interface {
 	Connect() error
 }
 
+func upsertCluster(t testing.NTB, cluster Cluster, opts *ntopts.New) {
+	var adoptCluster bool
+	switch *e2e.CreateClusters {
+	case e2e.CreateClustersEnabled:
+		adoptCluster = false
+	case e2e.CreateClustersLazy:
+		adoptCluster = true
+	default:
+		t.Fatalf("unrecognized value for create-cluster: %s", *e2e.CreateClusters)
+	}
+	exists, err := cluster.Exists()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if exists && !adoptCluster {
+		t.Fatalf("cluster %s already exists and create-clusters=%s", opts.ClusterName, *e2e.CreateClusters)
+	}
+	t.Cleanup(func() {
+		if !*e2e.DestroyClusters || (t.Failed() && *e2e.Debug) {
+			t.Logf("[WARNING] Skipping deletion of %s cluster %s", *e2e.TestCluster, opts.ClusterName)
```

Comment:
Maybe add (--delete-clusters=false) here to indicate that this was asked for behavior and not actually a problem/error.

Reply:
agreed, done